### PR TITLE
Remove label mapping for `service.instance.id`

### PR DESCRIPTION
**Pull request by zeitlinger**
- Resolves: https://github.com/open-telemetry/opentelemetry-operator/issues/3495

**Testing:** updated existing

**Documentation:** updated existing

### CODE REVIEW

File: .chloggen/service-instanc-id-mapping.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+note: Remove the mapping of `app.kubernetes.io/instance` to `service.instance.id`, because multiple containers 
+  should never share the same `service.instance.id`
+
+# One or more tracking issues related to the change
+issues: [3495]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
```
[the remark this answers is not in the document]
Comment (zeitlinger) [reply .chloggen/service-instanc-id-mapping.yaml]: done

File: .chloggen/service-instanc-id-mapping.yaml [hunk 2]
```diff
@@ -5,13 +5,23 @@ change_type: bug_fix
...
 issues: [3495]
 
 # (Optional) One or more lines of additional information to render under the primary note.
 # These lines will be padded with 2 spaces and then inserted directly into the document.
 # Use pipe (|) for multiline entries.
-subtext:
+subtext: |
+  Technically, this is a breaking change, but we regard it as a bug fix because the previous behavior was incorrect.
+  
+  In you did have multiple containers and use `app.kubernetes.io/instance` to set the `service.instance.id`,
```
Comment (jaronoff97) [inline .chloggen/service-instanc-id-mapping.yaml:19]:
```suggestion
  if you did have multiple container instrumentation and use `app.kubernetes.io/instance` to set the `service.instance.id`, 
```

File: .chloggen/service-instanc-id-mapping.yaml [hunk 3]
```diff
@@ -5,13 +5,23 @@ change_type: bug_fix
...
 # Use pipe (|) for multiline entries.
-subtext:
+subtext: |
+  Technically, this is a breaking change, but we regard it as a bug fix because the previous behavior was incorrect.
+  
+  In you did have multiple containers and use `app.kubernetes.io/instance` to set the `service.instance.id`, 
+  you will now see multiple instances in the UI - which is the correct behavior.
+  
+  You can still use the attribute `resource.opentelemetry.io/service.instance.id` to set the `service.instance.id`,
+  which will be shared across all containers in the pod - but this is not recommended.
```
Comment (jaronoff97) [inline .chloggen/service-instanc-id-mapping.yaml:23]:
```suggestion
  which will be shared across all containers in the pod - but this is not recommended for multiple container instrumentation instances.
```

File: README.md [hunk 4]
```diff
@@ -794,6 +794,35 @@ The priority for setting resource attributes is as follows (first found wins):
 This priority is applied for each resource attribute separately, so it is possible to set some attributes via
 annotations and others via labels.
 
+### How resource attributes are calculated from the pod's metadata
```
Comment (zeitlinger) [inline README.md:797]: @cyrille-leclerc please review is this is clear

**Comment by zeitlinger** [comment]
@jaronoff97 please check again

**Comment by jaronoff97** [comment]
@zeitlinger i want to think a bit more about this after some of the user feedback in the thread in slack. Maybe we should discuss this at our SIG meeting this week?

**Comment by zeitlinger** [comment]
@jaronoff97 as discussed in the meeting last week, this should be good to merge.

See https://docs.google.com/document/d/1Unbs2qp_j5kp8FfL_lRH-ld7i5EOQpsq0I4djkOOSL4/edit?tab=t.0#bookmark=id.bqi399f3q1ef

File: README.md [hunk 5]
```diff
@@ -794,6 +794,35 @@ The priority for setting resource attributes is as follows (first found wins):
...
+- `k8s.daemonset.name`
+- `k8s.cronjob.name`
+- `k8s.job.name`
+- `k8s.pod.name`
+- `k8s.container.name`
+
+#### How service.version is calculated
+
+Take the tag from the docker image name of the container.
+If the tag contains a `/`, the tag is ignored (this can happen if the image name contains a port number).
```
Comment (cyrille-leclerc) [inline README.md:817]:
```suggestion
- `pod.annotation[resource.opentelemetry.io/service.version]`
- `if (cfg[useLabelsForResourceAttributes]) pod.label[app.kubernetes.io/version]`
- `if (contains(container.image.tags[0], '/') == false) contains(container.image.tags[0]
```

Please check my pseudo code, I'm not super familiar with the array-ish nature of [`container.image.tags[]`](https://opentelemetry.io/docs/specs/semconv/attributes-registry/container/#container-image-tags)

**Review by cyrille-leclerc — COMMENTED** [review]
I added proposal to document with pseudo code the naming strategy as it's quite complex

**Review by swiatekm — APPROVED** [review]
:+1: Thank you for the updated documentation, it really helps.

**Review by jaronoff97 — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -794,6 +794,38 @@ The priority for setting resource attributes is as follows (first found wins):
...
+
+- `pod.annotation[resource.opentelemetry.io/service.version]`
+- `if (cfg[useLabelsForResourceAttributes]) pod.label[app.kubernetes.io/version]`
+- `if (contains(container docker image tag, '/') == false) container docker image tag`
+
+#### How `service.instance.id` is calculated
+                                   
+
+- `pod.annotation[resource.opentelemetry.io/service.instance.id]`
+- `if (config[useLabelsForResourceAttributes]) pod.label[app.kubernetes.io/instance]`
```
Comment (cyrille-leclerc) [inline README.md:826]: @zeitlinger shouldn't we remove this line from the docs?

**Pull request merged**